### ISSUE TITLE
Comments lost in translation

### ISSUE THREAD

**Issue by kobliha** [yast/ycp-killer#618]
Comments around `switch, case, default` are lost:

Compare:
- https://github.com/yast/yast-dns-server/blob/good-bye-ycp/src/dialog-masterzone.ycp#L958
- https://github.com/yast/yast-dns-server/blob/hello-ruby/src/include/dns-server/dialog-masterzone.rb#L1289

And:
- https://github.com/yast/yast-dns-server/blob/good-bye-ycp/src/dialog-masterzone.ycp#L1017
- https://github.com/yast/yast-dns-server/blob/hello-ruby/src/include/dns-server/dialog-masterzone.rb#L1380

**Comment by dmajda** [comment]
I created a minimal testcase:

**YCP**

``` ycp
{
  switch (1) {
    // case comment
    case 1:
      y2milestone("M1");
      break;
    // default comment
    default:
      y2milestone("M2");
  }
}
```

**Ruby**

``` ruby
# encoding: utf-8

module Yast
  class DefaultClient < Client
    def main
      case 1
        when 1
          Builtins.y2milestone("M1")
        else
          Builtins.y2milestone("M2")
      end

      nil
    end
  end
end

Yast::DefaultClient.new.main
```

The comments are missing from XML already, looks like another YCP parser bug. I'll dig deeper.

**Comment by dmajda** [comment]
The comments are lost because there is no place to attach them to — no `YCode` instance is generated for `case` and `default` clauses. They are represented only as indices into a statement list inside the `switch` block plus an associated `YCPValue` in case of `case`.

The only way out of this would be to change the representation. But that would not be trivial and it would have quite some regression potential. I don't think we should do it just because of the comments.

Now that I worked on this problem I _think_ we actually encountered this during Y2R development and decided we won't fix it, but I am not 100% sure.

While I was digging in the YCP parser, I encountered few things which I think are bugs:
1. The `*_COMMENT` macros in [`case` handling code](https://github.com/yast/yast-core/blob/0dda969d3ef29296226e7bc41e2bb3234661bc32/libycp/src/parser.yy#L1878) and [`default` handling code](https://github.com/yast/yast-core/blob/0dda969d3ef29296226e7bc41e2bb3234661bc32/libycp/src/parser.yy#L1941) are probably useless given the fact the comments are not saved.
2. The `RULE_COMMENT` macros in [`default` handling code](https://github.com/yast/yast-core/blob/0dda969d3ef29296226e7bc41e2bb3234661bc32/libycp/src/parser.yy#L1941) operate on `$$.c` which was set to `0` on the previous line. This should normally lead to `y2error` inside `attach_comment`, but it doesn't, because of the next problem.
3. No comments were present in `$1.com`, `$2.com`,... in `case` and `default` handling code when I executed the code on the testcase above. I'm not sure why.

Because of discovered problems, I'll leave the issue open and let @mvidner and @jreidinger to investigate further once they are back from vacation. Waiting shouldn't do any harm, as the discovered problems taken together don't affect YCP parser behavior and my conclusion (this is not worth fixing) likely won't change.
